external/llvm-project/mlir/lib/Conversion/AMDGPUToROCDL/AMDGPUToROCDL.cpp

```diff
@@ -401,28 +402,30 @@ static void wmmaPushInputOperand(ConversionPatternRewriter &rewriter,
   if (elemType.isBF16())
     llvmInput = rewriter.create<LLVM::BitcastOp>(
         loc, vectorType.clone(rewriter.getI16Type()), llvmInput);
-  if (!elemType.isInteger(8)) {
+  if (elemType.getIntOrFloatBitWidth() != 8) {
     operands.push_back(llvmInput);
     return;
   }
 
+  auto mlirInputType = dyn_cast<VectorType>(mlirInput.getType());
+  if (mlirInputType.getElementType().isInteger(8)) {
+    // if element type is 8-bit signed or unsigned, ignore the isUnsigned flag
+    bool localIsUnsigned = isUnsigned;
+    if (elemType.isUnsignedInteger(8)) {
+      localIsUnsigned = true;
+    } else if (elemType.isSignedInteger(8)) {
+      localIsUnsigned = false;
+    }
+    Value sign = createI1Constant(rewriter, loc, !localIsUnsigned);
+    operands.push_back(sign);
```

Comment:
If you are flipping sign with `!localIsUnsigned` then perhaps better to make it `localIsSigned`

Reply:
So the point is that the wmma Operation has a `getUnsigned()` method. So at some point we need to to that flip to reconcile LLVM vs MLIR operations.

Reply:
More generally, the WMMA op has been rigged to work whether you specify the signedness on the type or on the operation (the latter is more preferred, but I left the type-level version in case people want to go that route)